### PR TITLE
Fix power requirements rounding issue

### PR DESCRIPTION
**Pull request by samsmithnz**
Update power calculation to avoid rounding issues for buildings with low power.

* Change `totalPower` property in `Factory` and `BuildingRequirement` interfaces in `web/src/interfaces/planner/FactoryInterface.ts` to be a number instead of a string.
* Modify `calculateBuildingRequirements` function in `web/src/utils/factory-management/buildings.ts` to use precise floating-point arithmetic for calculating total power.
* Update `calculateBuildingsAndPower` function in `web/src/utils/factory-management/buildings.ts` to use floating-point arithmetic for summing total power requirements.
* Update `totalPower` property in demo factories and `buildingRequirements` in `web/src/utils/demoFactories.ts` to be a number instead of a string.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/Maelstromeous/satisfactory-factories/pull/36?shareId=1094393e-bc1d-4579-9857-865328247e15).

### CODE REVIEW

**Comment by Maelstromeous** [comment]
fyi I should have fixed vercel not allowing deployments without autorisation, any more commits should automatically generate the preview now.

**Comment by Maelstromeous** [comment]
Failed with interface issues:

```
src/utils/demoFactories.ts(12,50): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(20,50): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(64,63): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(95,49): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(121,61): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(144,53): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(152,53): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(187,69): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(213,53): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(239,69): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(262,51): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(304,7): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(331,54): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/demoFactories.ts(382,7): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/factory-management/factory.ts(28,5): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/factory-management/mocks/mockRecipes.ts(24,7): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/factory-management/mocks/mockRecipes.ts(49,7): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/factory-management/mocks/mockRecipes.ts(74,7): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/factory-management/mocks/mockRecipes.ts(99,7): error TS2322: Type 'string' is not assignable to type 'number'.
src/utils/factory-management/mocks/mockRecipes.ts(130,7): error TS2322: Type 'string' is not assignable to type 'number'.
 ELIFECYCLE  Command failed with exit code 2.
Error: Command "pnpm build" exited with 2
```

**Comment by Maelstromeous** [comment]
Closes #5

File: web/src/utils/factory-management/buildings.ts [hunk 1]
```diff
@@ -36,27 +36,26 @@ export const calculateBuildingRequirements = (factory: Factory, gameData: DataIn
...
         name: building.name,
-        amount: '0',
+        amount: 0,
         powerPerBuilding: building.powerPerBuilding,
-        totalPower: '0',
+        totalPower: 0,
       }
     }
 
     const facBuilding = factory.buildingRequirements[building.name]
```
Comment (samsmithnz) [inline web/src/utils/factory-management/buildings.ts:54]: @Maelstromeous - what does this constant do? I suspect it can be removed and is unused.

Comment (samsmithnz) [reply web/src/utils/factory-management/buildings.ts:54]: For clarity, these lines can be removed I think:

```
    const facBuilding = factory.buildingRequirements[building.name]

    facBuilding.amount = facBuilding.amount + building.amount
    facBuilding.totalPower = facBuilding.totalPower + building.totalPower
```

Comment (Maelstromeous) [reply web/src/utils/factory-management/buildings.ts:54]: It's a reference to `factory.buildingRequirements[constructor], just a shortcut really for factory.buildingRequirements[consturctor].amount etc.

They're used for calculating the factory's building requirements, and I believe is very much needed.

**Comment by samsmithnz** [comment]
I think I need to throw this away and start over. Merging in the changes from main has broken a few things.

**Comment by samsmithnz** [comment]
was this fixed on Main? lol. It's working for me now. 🤣

**Comment by samsmithnz** [comment]
I see what I did. Silly branch merges. I don't often fork, so that added a new wrinkle to the chaos.

**Comment by samsmithnz** [comment]
Closing this in favor of a new PR, #40

**Pull request closed without merging**